tests/Imgix/Tests/UrlHelperTest.php

```diff
@@ -4,6 +4,65 @@
 
 class UrlHelperTest extends \PHPUnit\Framework\TestCase {
 
+    /*--- formatPath() ---*/
+    public function testHelperFormatPathWithSimplePath() {
+        $path = "dog.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/" . $path);
+    }
+
+    public function testHelperFormatPathWithLeadingSlash() {
+        $path = "dog.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath("/" . $path), "/" . $path);
+    }
+
+    public function testHelperFormatPathWithUnencodedCharacters() {
+        $path = "images/püg.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/images/p%C3%BCg.jpg");
+    }
+
+    public function testHelperFormatPathWithFullUrl() {
+        $path = "http://mywebsite.com/images/dog.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/http%3A%2F%2Fmywebsite.com%2Fimages%2Fdog.jpg");
+    }
+
+    public function testHelperFormatPathWithFullHttpsUrl() {
+        $path = "https://mywebsite.com/images/dog.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/https%3A%2F%2Fmywebsite.com%2Fimages%2Fdog.jpg");
+    }
+
+    public function testHelperFormatPathWithFullUrlWithUnencodedCharacters() {
+        $path = "http://mywebsite.com/images/püg.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/http%3A%2F%2Fmywebsite.com%2Fimages%2Fp%C3%BCg.jpg");
+    }
+
+    public function testHelperFormatPathWithFullUrlWithLeadingSlash() {
+        $path = "/http://mywebsite.com/images/dog.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        $this->assertEquals($uh->formatPath($path), "/http%3A%2F%2Fmywebsite.com%2Fimages%2Fdog.jpg");
+    }
+
+    public function testHelperFormatPathWithFullUrlWithEncodedCharacters() {
+        $path = "http://mywebsite.com/images/p%C3%BCg.jpg";
+        $uh = new URLHelper("test.imgix.net", $path);
+
+        // The pre-encoded characters should now be *double* encoded
+        $this->assertEquals($uh->formatPath($path), "/http%3A%2F%2Fmywebsite.com%2Fimages%2Fp%25C3%25BCg.jpg");
```

Comment:
Why would we want to encode an already encoded path?

Reply:
It's basically impossible to detect whether the path we're given has been pre-encoded or not, so we need to assume all paths are given to us *unencoded* and encode them blindly. This test just makes sure that, in the rare event that someone chooses to pass in a pre-encoded path, it gets double-encoded as expected. More of a safeguard to make sure we don't break this edge-case behavior in the future by accident.

Reply:
That makes sense. Thanks for clarifying.